### ISSUE TITLE
Test node creation error with Firefox

### ISSUE THREAD

**Issue by j-pillon** [campsych/concerto-platform#25]
How to reproduce:
- OS: Windows 7 sp1
- Use Firefox 45.2 ESR as web browser
- Add new test or edit existing test
- Right click in "Test flow" section to show context menu
- Click "create_template_definition" (or other)

Results:
- No node created
- SQL error in prod.log
```
[2016-07-22 12:52:25] request.CRITICAL: Uncaught PHP Exception Doctrine\DBAL\Exception\NotNullConstraintViolationException: "An exception occurred while executing 'INSERT INTO TestNode (type, posX, posY, updated, created, tags, flowTest_id, sourceTest_id, updatedBy_id) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?)' with params ["0", null, null, "2016-07-22 12:52:25", "2016-07-22 12:52:25", "", 30, 8, null]:  SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'posX' cannot be null" at D:\Program Files\wamp\www\concerto\vendor\doctrine\dbal\lib\Doctrine\DBAL\Driver\AbstractMySQLDriver.php line 112 {"exception":"[object] (Doctrine\\DBAL\\Exception\\NotNullConstraintViolationException(code: 0): An exception occurred while executing 'INSERT INTO TestNode (type, posX, posY, updated, created, tags, flowTest_id, sourceTest_id, updatedBy_id) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?)' with params [\"0\", null, null, \"2016-07-22 12:52:25\", \"2016-07-22 12:52:25\", \"\", 30, 8, null]:\n\nSQLSTATE[23000]: Integrity constraint violation: 1048 Column 'posX' cannot be null at D:\\Program Files\\wamp\\www\\concerto\\vendor\\doctrine\\dbal\\lib\\Doctrine\\DBAL\\Driver\\AbstractMySQLDriver.php:112, Doctrine\\DBAL\\Driver\\PDOException(code: 23000): SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'posX' cannot be null at D:\\Program Files\\wamp\\www\\concerto\\vendor\\doctrine\\dbal\\lib\\Doctrine\\DBAL\\Driver\\PDOStatement.php:93, PDOException(code: 23000): SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'posX' cannot be null at D:\\Program Files\\wamp\\www\\concerto\\vendor\\doctrine\\dbal\\lib\\Doctrine\\DBAL\\Driver\\PDOStatement.php:91)"} []
```

Analyse:
- posX and posY columns cannot be NULL
- The functions "window.rightClickEvent.offsetX" and "window.rightClickEvent.offsetY" return undefined.
- Firefox js functions for mouse cursor position aren't well implemented or missing (offsetX and offsetY -> \src\PanelBundle\Resources\public\angularjs\app\concerto_panel\js\directives\flow_logic_directive.js).

Solution:
- First, using Chrome to test other functionalies.
- Special functions have to be implemented to support Firefox (or maybe initialize PosX and PosY to the top left position)

bye

**Comment by przemyslawlis** [comment]
Was not able to reproduce it on both **FF 45.2 ESR** and **FF 47.0.1**.
Are you on latest **master** branch source? If not, can you please recheck on latest?
Also please try to clear your browser cache, and Symfony cache (remove all directories with their content inside **/app/cache**) before rechecking just in case.

**Comment by j-pillon** [comment]
Sorry, i'm guilty...

I've 2 versions of Firefox (38.6 ESR x86 and 45.2 ESR x64) on my computer. Of course, the shortcut target the 38.6 version... offsetX and offsetY functions have been implemented only since Firefox 39.
Tested perfect on FF 45.2 ESR x64.

**Comment by przemyslawlis** [comment]
Good find!
Reproduced on **FF 38.6 ESR**.
Fixed in 26be4b589158fd232847b1326b15c7a169ff61fc
